sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java

```diff
@@ -883,6 +913,9 @@ private void validateConfig() {
                     LocationHelper.getLocationEndpoint(uri, trimmedPreferredRegion);
                 }
             );
+
+            ifThrowIllegalArgException(this.proactiveContainerInitConfig.getNumProactiveConnectionRegions() > this.preferredRegions.size(), "no. of regions to proactively connect to" +
```

Comment:
Possible NullPointerException?

```suggestion
            ifThrowIllegalArgException(this.proactiveContainerInitConfig != null && this.proactiveContainerInitConfig.getNumProactiveConnectionRegions() > this.preferredRegions.size(), "no. of regions to proactively connect to" +
```

Reply:
this.preferredRegiosn could also be null - if proactiveContainerInitConfig != null && preferredRegions == null we shoudl also throw - we want to encourage setting preferred regions